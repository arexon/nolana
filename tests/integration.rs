use std::{fs, path::Path};

use insta::Settings;
use nolana::Parser;

fn with_settings(f: impl FnOnce()) {
    let mut settings = Settings::clone_current();
    settings.set_omit_expression(true);
    settings.set_prepend_module_to_snapshot(false);
    settings.bind(f)
}

fn read_and_parse(path: &Path) -> String {
    let source = fs::read_to_string(path).unwrap();
    let result = Parser::new(&source).parse();
    format!("{result:#?}")
}

#[test]
fn test_parser() {
    with_settings(|| {
        insta::glob!("parser/*.nolana", |path| {
            insta::assert_snapshot!(read_and_parse(path));
        });
    });
}
