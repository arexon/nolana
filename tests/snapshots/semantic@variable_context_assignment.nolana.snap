---
source: tests/integration.rs
input_file: tests/semantic/variable_context_assignment.nolana
---
[
    Diagnostic {
        inner: DiagnosticInner {
            message: "`context.` variables are read-only",
            labels: Some(
                [
                    LabeledSpan {
                        label: None,
                        span: SourceSpan {
                            offset: SourceOffset(
                                0,
                            ),
                            length: 15,
                        },
                        primary: false,
                    },
                ],
            ),
            help: Some(
                "Try assigning to `variable.` or `temp.` instead",
            ),
            severity: Error,
        },
    },
]
